src/Data/Vec/Binary/Base.agda

```diff
@@ -0,0 +1,95 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Vectors with faster indexing based on binary numbers
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Vec.Binary.Base where
+
+open import Data.Fin.Binary.Base
+open import Data.Nat.Binary.Base
+open import Level using (Level)
+
+private
+  variable
+    a b c : Level
+    A : Set a
+    B : Set b
+    C : Set c
+    n : ℕᵇ
+
+-- Core type
+------------------------------------------------------------------------
+
+data Vecᵇ (A : Set a) : ℕᵇ → Set a where
+  -- The empty vector
+  []        :                               Vecᵇ A zero
+  -- A vector with an odd number of elements
+  _∷⟨_/_⟩   : A     → Vecᵇ A n → Vecᵇ A n → Vecᵇ A 1+[2 n ]
+  -- A vector with a non-zero even number of elements
+  _×_∷⟨_/_⟩ : A → A → Vecᵇ A n → Vecᵇ A n → Vecᵇ A 2[1+ n ]
```

Comment:
I would be tempted to use two `Vecᵇ A (suc n)` instead?

Reply:
That feels like it would complicate a lot of definitions for not much improvement. Why does it tempt you?

Reply:
Probably because I don't have a global view of the design and this thought was from looking just at the definition here on its own.

So this probably calls for (more) comments rather than an actual change.

Reply:
It's difficult for me to explain when I don't know what's not being understood! 

One thing that might be making this more confusing is that `Vecᵇ` is indexed by a `ℕᵇ`, not a more familiar `ℕ`. This means that `suc` is a function, not a constructor, and `suc n` isn't structurally smaller than `2[1+ n ]`.

Reply:
I think you are correct that this indexing is definitely part of the reason I'm not quite understanding. Maybe if more of the semantics of things were explained, it would all make more sense.